numpyro/compat/util.py

```diff
@@ -0,0 +1,7 @@
+class UnsupportedAPIWarning(Warning):
```

Comment:
Looks great! If this class proves useful, we may want to move it to `pyroapi`, maybe calling it a `NotImplementedWarning` to match `NotImplementedError`.